xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/editactions.vm

```diff
@@ -27,4 +27,25 @@
     #editActionButton($saveaction, 'saveandview', 'btn-success')
     #editActionButton('cancel', 'cancel', 'btn-default')
   </div> ## buttons
+  ## Include the overwrite modal dialog that might be needed when copying a template's document hierarchy.
+  #if ("$!request.template" != '')
+    <div class="modal" id="overwriteQuestionModal" tabindex="-1" role="dialog" aria-labelledby="overwriteQuestionModalLabel">
+      <div class="modal-dialog" role="document">
+        <div class="modal-content">
+          <div class="modal-header">
+            <h4 class="modal-title" id="overwriteQuestionModalLabel">$services.localization.render('core.editors.save.overwrite.title')</h4>
+          </div>
+          <div class="modal-body">
+            <p>$services.localization.render('core.editors.save.overwrite.destinationExists', ['<span class="overwrite-question-destination"></span>'])</p>
+            <p>$services.localization.render('core.editors.save.overwrite.confirmWithSource', ['<span class="overwrite-question-source"></span>'])</p>
+          </div>
+          <div class="modal-footer">
+            <label class="checkbox-inline"><input type="checkbox"/>$services.localization.render('core.editors.save.overwrite.dontAskAgain')</label>
+            <button type="button" class="btn btn-default" data-dismiss="modal" data-overwrite="true">$services.localization.render('yes')</button>
+            <button type="button" class="btn btn-primary" data-dismiss="modal" data-overwrite="false">$services.localization.render('no')</button>
+          </div>
+        </div>
+      </div>
+    </div>
+  #end
```

Comment:
Note that the `overwritequestion.vm` below can also be reused in the other actions that need a generic overwrite question to be answered in the UI.